### PR TITLE
fix: node compat

### PR DESCRIPTION
**Pull request by ronag**

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -7,12 +7,21 @@ function shift (stream) {
 }
 
 function getStateLength (state) {
+  if (state.readableLength != null) {
+    return state.readableLength
+  }
+
+  if (state.length != null) {
+    return state.lenght
```
Comment (mafintosh) [inline index.js]: typo here

Comment (mafintosh) [reply index.js]: The two lengths here refer to the next chunk in the buffer?

Comment (ronag) [reply index.js]: That was wrong. Updated.

**Comment by ronag** [comment]
I'm actually unsure what this module does differently from just doing a `.read()`?

**Comment by mafintosh** [comment]
Screenshot 2024-01-17 at 11 34 45
@ronag docs say it returns all the data concatted if no size is provided (havent tested myself in a while so unsure). The point of this is to return the next buffer chunk that was pushed to the stream. if thats what the stream does indeed then yea obvs easier to do that.

ie

```js
stream.push(Buffer.from('a'))
stream.push(Buffer.from('b'))

shift(stream) // buffer('a')
shift(stream) // buffer('b')
```

**Comment by ronag** [comment]
I will double-check and get back to you!

**Comment by ronag** [comment]
As far as i can tell, if you don't pass anything into read it will give you the first buffer:

https://github.com/nodejs/node/blob/main/lib/internal/streams/readable.js#L634

But it's not entirely clear. I think it only applies if it's "flowing".

**Comment by ronag** [comment]
This really needs some review, either way, in my opinion `read()` should always return the first buffer. If that's not the case we should probably fix it. Either way, I'm pretty sure the current state of this PR will make things better.

**Comment by mafintosh** [comment]
It doesnt for me in node 20. ie

```js
const { Readable } = require('stream')

const rs = new Readable({ read () { } })

rs.push(Buffer.from('a'))
rs.push(Buffer.from('b'))

console.log(rs.read())
```

Prints `<Buffer 61 62>`

**Comment by mafintosh** [comment]
Thanks though @ronag , merging this for now.

**Comment by mafintosh** [comment]
1.0.3

**Comment by nadavzipo** [comment]
@mafintosh any chance you bumping https://github.com/mafintosh/duplexify to use `1.0.3`?
It will fix https://github.com/googleapis/nodejs-storage/issues/2391 issue.
Thanks!

**Pull request merged**